test/typescript/index.js

```diff
@@ -0,0 +1,52 @@
+"use strict";
```

Comment:
Warning -- we have transpiled code here I'm guessing we don't want.

Reply:
Would you rather we remove this? I wanted to keep it as a snapshot of sorts for our TS tests, but I can see the argument for removing it, updating the test-ts-usage script, and just expecting the test to pass instead of comparing the transpiled code?

Reply:
yeah we should 100% .gitignore this. i think the best way is to add to the project .gitignore (and not create one just for this dir). i should be something like `test/typescript/*.js` - i.e. ignore _all_ JS files.

> I wanted to keep it as a snapshot of sorts for our TS tests

The `test/typescript/index.tsx` file (technically combined with TS versions and tsconfig) will be a good enough snapshot.